website_sale_wishlist/models/product_wishlist.py

```diff
@@ -68,10 +68,14 @@ def _join_current_user_and_session(self):
         user_products = self.search([
             ("user_id", "=", self.env.uid),
         ]).mapped("product_tmpl_id")
-        session_domain = [
-            ("session", "=", request.session.sid),
-            ("user_id", "=", False),
-        ]
+
+        if not request:
+            session_domain = [("user_id", "=", False)]
```

Comment:
You'd be assigning all dangling products from all dangling sessions as favourites for this user when logging in without session.

If there's no session, just skip this whole method, since it makes no sense in such circumstance.